developer-guides/miscellaneous/multi-message-transaction.md

```diff
@@ -0,0 +1,281 @@
+            "amount": "1000000"
+          }
+        ]
+      },
+      {
+        "@type": "/cosmos.bank.v1beta1.MsgSend",
+        "from_address": "juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y",
+        "to_address": "juno1ka8v934kgrw6679fs9cuu0kesyl0ljjy4tmycx",
+        "amount": [
+          {
+            "denom": "ujuno",
+            "amount": "5000000"
+          }
+        ]
+      }
+    ],
+    "memo": "",
+    "timeout_height": "0",
+    "extension_options": [],
+    "non_critical_extension_options": []
+  },
+  "auth_info": {
+    "signer_infos": [],
+    "fee": {
+      "amount": [
+          {
+              "denom": "ujunox",
+              "amount": "5000"
+          }
+      ],
+      "gas_limit": "200000",
+      "payer": "",
+      "granter": ""
+    }
+  },
+  "signatures": []
+}
+```
+
+In this case, we send 5JUNO to juno1ka... as well as the original account, at the same time. (Reminder: ujuno is a 6th exponent representation of human readable form). So in the above messages, we in total send 6JUNO.&#x20;
+
+**NOTE**: If our account only has less than 6JUNO, the entire above transaction would fail (non 0 return code) for not enough balance. Even through the first message is only 1JUNO, all messages in a transaction do not execute unless they are all successful. This is a safety mechanism!
+
+Then we can sign and broadcast it
+
+```bash
+# Sign the message from the account who is sending the tokens
+# This creates a new file which adds a signature to the signatures array
+junod tx sign bankmsgs.json \
+    --chain-id=uni-6 --from KEY [flags] &> signed_bankmsgs.json
+
+# Broadcast the message to the chain after it is successfully signed
+junod tx broadcast signed_bankmsgs.json \
+    --from KEY --chain-id=uni-6 [flags]
+
+# Query to ensure it went through well (enough gas and fees)
+junod q tx 12A4D888F6737130CE03CEC6BBAFD98855462E81D5AF871949E1EDE1B0B1A14B --node https://uni-rpc.reece.sh:443
+```
+
+## Multi Message NFT Mint (CosmWasm)
+
+After reading the above guide, the same steps can be applied to mint cosmwasm based NFTs (CW721s). This requires you to submit an NFT contract to chain first, and setup with your collections values before moving on to this step. We will link to that guide HERE in the future.
+
+```sh
+# generate the initial Transaction structure
+junod tx wasm execute juno1za0uemnhzwkjrqwguy34w45mqdlzfm9hl4s5gp5jtc0e4xvkrwjs6s2rt4 \
+    '{"mint":{"token_id":"1","owner":"juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y","token_uri":"https://domain.com/image1.png"}}' \
+    --output json -y --generate-only --from juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y
+```
+
+```json
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/cosmwasm.wasm.v1.MsgExecuteContract",
+        "sender": "juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y",
+        "contract": "juno1za0uemnhzwkjrqwguy34w45mqdlzfm9hl4s5gp5jtc0e4xvkrwjs6s2rt4",
+        "msg": {
+            "mint": {
+                "token_id": f"1",
```

Comment:
not sure what f is in > `f"1",` ?

Reply:
ahh, I had fstrings here from python, removing